### PR TITLE
P2 202 fe start payroll

### PR DESCRIPTION
**Pull request by ALFAROSO**
payroll-1

[Jira](https://team-voyagers.atlassian.net/jira/software/projects/P2/boards/2?assignee=712020%3Aa4d78f0d-920a-427a-a5ce-b36979797950&selectedIssue=P2-202)

### CODE REVIEW

File: src/pages/Payroll/Payroll.tsx [hunk 1]
```diff
@@ -1,96 +1,191 @@
...
-const columns : GridColDef[] = [
+import { Select, MenuItem } from "@mui/material";
+import { useNavigate } from "react-router-dom";
+import React from "react";
+
+const columns: GridColDef[] = [
   {
     field: "period",
-    headerName: "Pay Period",
+    headerName: "PAY PERIOD",
```
Comment (whamester) [inline src/pages/Payroll/Payroll.tsx:20]: The capitalization should be on the styles, you can leave the text as it was

File: src/pages/Preview/Preview.tsx [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+    field: "netAmount",
+    headerName: "Net Pay ($)",
+    width: 150,
+    renderCell: (params) => {
+      return <span>{params.row.netAmount}</span>;
+    },
+  },
+];
+
+function formatDate(value: Dayjs): string {
```
Comment (whamester) [inline src/pages/Preview/Preview.tsx:81]: remove this function and use intl components

File: src/pages/Preview/Preview.tsx [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+            alignItems: "center",
+            justifyContent: "space-between",
+          }}
+        >
+          <LocalizationProvider dateAdapter={AdapterDayjs}>
+            <DemoContainer components={["DatePayroll"]}>
+              <span>Select the date range:</span>
+              <span>From</span>
+              <DatePicker
+                // label="Start Date"
```
Comment (whamester) [inline src/pages/Preview/Preview.tsx:241]: remove comment

File: src/pages/Preview/Preview.tsx [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+      <Box display="flex" flexDirection="column" flexGrow={1} pb={3}>
+        <Grid container spacing={2}>
+          <Grid item xs={3}>
+            <Typography variant="h6" gutterBottom>
+              PAY PERIOD
+            </Typography>
+          </Grid>
+          <Grid item xs={3}>
+            <Typography variant="h6" gutterBottom>
+              TOTAL NET PAY
```
Comment (whamester) [inline src/pages/Preview/Preview.tsx:310]: uppercase should be on the design, we should not use Typography, but in this case Label from our custom components

File: src/pages/Payroll/Payroll.tsx [hunk 5]
```diff
@@ -1,96 +1,191 @@
...
+      }
+    };
+  
+    fetchSeasons();
+  }, []);
 
-    const { user } = useUser();
+  let uniqueFarmId: string[] = [];
+  if (data && data.length > 0) {
+    uniqueFarmId = [...new Set((data as any[]).map((item: any) => item.farm))];
```
Comment (whamester) [inline src/pages/Payroll/Payroll.tsx:151]: you should get the farmId with

```
const { user } = useUser()

user.farm.id

```

File: src/pages/Payroll/Payroll.tsx [hunk 6]
```diff
@@ -109,8 +204,36 @@ function formatDate(
           href: "",
         },
       ]}
-      actions={<Button variant="contained" endIcon={<CaretRight />}>Start a payroll</Button>}
+      actions={
+        <Button
+          variant="contained"
+          onClick={() =>
+            navigate("/payroll/preview", {
+              state: { uniqueFarmId, uniqueSeasonId , uniqueSeasonNa},
```
Comment (whamester) [inline src/pages/Payroll/Payroll.tsx:212]: pass the parameters, with query params, and don't pass the season name, there should be a request to get the season information with the id on the preview

Comment (ALFAROSO) [reply src/pages/Payroll/Payroll.tsx:214]: The endpoint on the preview page does not return the name, and we need it for the modal

Comment (ALFAROSO) [reply src/pages/Payroll/Payroll.tsx]: P2-280-add-season-name-on-preview-endpoint-be

**Review by whamester — APPROVED** [review]

**Review by ReinhardtBotha — APPROVED** [review]
MonkeyMoneyGIF

**Review by BryanCaldeira — APPROVED** [review]

**Pull request merged**